qa/rpc-tests/test_framework/util.py

```diff
@@ -398,6 +439,83 @@ def random_transaction(nodes, amount, min_fee, fee_increment, fee_variants):
 
     return (txid, signresult["hex"], fee)
 
+def split_transaction(node, prevouts, toAddrs, txfeePer=DEFAULT_TX_FEE_PER_BYTE,**kwargs):
+    """
+      Create a transaction that divides the sum of all the passed utxos into all the destination addresses
+      pass "sendtx=False" if you don't want to transaction to be submitted.
+      Returns (transaction in hex, Vin list, Vout list)
+    """
+    txid = None
+    inp = []
+    decContext = decimal.getcontext().prec
+    try:
+      decimal.getcontext().prec = 8
+      amount = Decimal(0)
+      # pdb.set_trace()
+      for tx in prevouts:
+        inp.append({"txid":str(tx["txid"]),"vout":tx["vout"]})
+	  # inp.append({"txid":bitcoin.core.b2lx(tx["outpoint"].hash),"vout":tx["outpoint"].n})
+	amount += tx["amount"]*BTC
+   
+      txLen = (len(prevouts)*100) + (len(toAddrs)*100)  # Guess the tx Size
+
+      while 1:
+	  outp = {} # = { str(toAddr): str((amount-txfeePer)/BTC) }
+          if amount-Decimal(txfeePer*txLen) < 0:  # fee too big, find something smaller
+            txfeePer = (float(amount)/txLen)/1.5
+          
+          txfee = int(math.ceil(txfeePer*txLen))
+	  amtPer = (Decimal(amount-txfee)/len(toAddrs)).to_integral_value()
+	  # print "amount: ", amount, " amount per: ", amtPer, "from :", len(prevouts), "to: ", len(toAddrs), "tx fee: ", txfeePer, txfee
+
+	  for a in toAddrs[0:-1]:
+	    if PerfectFractions:
+	      outp[str(a)] = str(amtPer/BTC)
+	    else:
+	      outp[str(a)] = float(amtPer/BTC)
+
+	  a = toAddrs[-1]
+	  amtPer = (amount - ((len(toAddrs)-1)*amtPer)) - txfee
+	  # print "final amt: ", amtPer
+	  if PerfectFractions:
+	      outp[str(a)] = str(amtPer/BTC)
+	  else:
+	      outp[str(a)] = float(amtPer/BTC)
+
+    #      try:
```

Comment:
I don't understand the exact question, but I pulled this code from prior code I wrote and I clearly see it needs a lot of cleanup so I cleaned it up.